client/src/components/mapping/ConceptMap/index.tsx

```diff
@@ -0,0 +1,569 @@
+
+  const createConceptMap = () => {
+    // TODO complete fhir concept map (there are missing fields we could fill)
+    const elements = conceptMap.reduce(
+      (acc: any[], row: MapRow) => [
+        ...acc,
+        {
+          code: row.source,
+          target: [{ code: row.target, equivalence: row.equivalence }]
+        }
+      ],
+      []
+    );
+
+    return {
+      name: conceptMapName, // for computer
+      title: conceptMapTitle, // for human
+      description: conceptMapDescription,
+      sourceUri: '', // value set uri
+      targetUri: '', // value set uri
+      group: [
+        {
+          source: sourceSystemName, // TODO make uri from that
+          target: selectedTargetSystem, // TODO make uri from that
+          element: elements
+        }
+      ]
+    };
+  };
+
+  const metaData = () => (
+    <div className="meta-information">
+      <div className="row">
+        <h2 className="left">Title</h2>
+        {!existingConceptMapId || modifyAnyway ? (
+          <InputGroup
+            className="right"
+            onChange={(e: React.FormEvent<HTMLElement>): void => {
+              const target = e.target as HTMLInputElement;
+              setConceptMapTitle(target.value);
+            }}
+            placeholder="Concept map title..."
+            value={conceptMapTitle}
+            small={true}
+          />
+        ) : (
+          conceptMapTitle
+        )}
+      </div>
+      <div className="row">
+        <h2 className="left">Name</h2>
+        {!existingConceptMapId || modifyAnyway ? (
+          <InputGroup
+            className="right"
+            onChange={(e: React.FormEvent<HTMLElement>): void => {
+              const target = e.target as HTMLInputElement;
+              setConceptMapName(target.value);
+            }}
+            placeholder="Concept map name..."
+            value={conceptMapName}
+            small={true}
+          />
+        ) : (
+          conceptMapName
+        )}
+      </div>
+      <div className="row">
+        <h2 className="left">Description</h2>
+        {!existingConceptMapId || modifyAnyway ? (
+          <InputGroup
+            className="right"
+            onChange={(e: React.FormEvent<HTMLElement>): void => {
+              const target = e.target as HTMLInputElement;
+              setConceptMapDescription(target.value);
+            }}
+            placeholder="Concept map description..."
+            value={conceptMapDescription}
+            small={true}
+          />
+        ) : (
+          conceptMapDescription
+        )}
+      </div>
+    </div>
+  );
+
+  const onSubmit = async (
+    e: React.FormEvent<HTMLFormElement>
+  ): Promise<void> => {
+    e.preventDefault();
+    if (creatingNewCodeSystem) {
+      const codeSystem = createCodeSystem();
+      try {
+        await axios.post(`${FHIR_API_URL}/CodeSystem`, codeSystem);
+        console.log('created code system', codeSystem);
+      } catch (err) {
+        console.error(
```

Comment:
can you display a toaster or something like that here ? we shouldn't have to open the console when this fails